two/two.cpp

```diff
@@ -214,14 +216,41 @@ bool Two::getCheck(SixPyObject *py_class, const char *init_config_str, const cha
     // create `args` and `kwargs` to invoke `AgentCheck` constructor
     args = PyTuple_New(0);
     kwargs = PyDict_New();
+    name = PyString_FromString(check_name);
+    PyDict_SetItemString(kwargs, "name", name);
     PyDict_SetItemString(kwargs, "init_config", init_config);
     PyDict_SetItemString(kwargs, "instances", instances);
 
     // call `AgentCheck` constructor
     check = PyObject_Call(klass, args, kwargs);
     if (check == NULL) {
-        setError("error creating check instance: " + _fetchPythonError());
-        goto done;
+        // clean any error from previous call
+        _fetchPythonError();
+
+        // TODO: add/return warning to the agent
```

Comment:
we can provide a different method through the api, like `get_check` and `get_check_deprecated` so the Agent would know which one failed